mono/metadata/metadata.c

```diff
@@ -7824,3 +7826,262 @@ mono_type_set_amods (MonoType *t, MonoAggregateModContainer *amods)
 	g_assert (t_full->mods.amods == NULL);
 	t_full->mods.amods = amods;
 }
+
+static void
+mono_signature_append_class_name (GString *res, MonoClass *klass)
+{
+	if (!klass) {
+		g_string_append (res, "<UNKNOWN>");
+		return;
+	}
+	if (klass->nested_in) {
+		mono_signature_append_class_name (res, klass->nested_in);
```

Comment:
```suggestion
		mono_signature_append_class_name (res, m_class_get_nested_in (klass));
```

We try not to access the fields of `MonoClass` directly (see `mono/metadata/class-getters.h`).